src/host/node_connections.h

```diff
@@ -344,6 +348,15 @@ namespace asynchost
       TCP s;
       s->set_behaviour(std::make_unique<OutgoingBehaviour>(*this, node));
 
+      if (client_host.has_value())
```

Comment:
Should/could this binding to the client host be more integrated in the existing TCP flow? If the binding fail, should we retry? 

In fact, just writing this comment makes me realise that this connection would proceed even if `uv_tcp_bind` fails, so I should integrate this in the existing flow better.

Reply:
Yes! :)

Reply:
Done.